contracts/finality/src/exec/finality.rs

```diff
@@ -445,15 +444,21 @@ pub(crate) mod tests {
         let add_finality_signature = get_add_finality_sig();
         let proof = add_finality_signature.proof.unwrap();
 
-        let initial_height = pr_commit.start_height;
+        // Convert the PubRandCommit in the type defined in this contract
+        let pr_commit = PubRandCommit {
+            start_height: pr_commit.start_height,
+            num_pub_rand: pr_commit.num_pub_rand,
+            height: pr_commit.height,
+            commitment: pr_commit.commitment,
+        };
 
         // Verify finality signature
         if proof.index < 0 {
             panic!("Proof index should be non-negative");
         }
```

Comment:
Comment in passing: These should be errors, not panics, in general.

Reply:
This is inside the unit test so replaced it with an assert now